CoreInteractiveGraph/test/unit/src/au/gov/asd/tac/constellation/graph/interaction/animation/AnimationNGTest.java

```diff
@@ -0,0 +1,271 @@
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package au.gov.asd.tac.constellation.graph.interaction.animation;
+
+import au.gov.asd.tac.constellation.graph.WritableGraph;
+import au.gov.asd.tac.constellation.graph.locking.DualGraph;
+import au.gov.asd.tac.constellation.graph.schema.SchemaFactoryUtilities;
+import au.gov.asd.tac.constellation.graph.schema.visual.VisualSchemaFactory;
+import au.gov.asd.tac.constellation.graph.schema.visual.concept.VisualConcept;
+import au.gov.asd.tac.constellation.utilities.camera.Camera;
+import au.gov.asd.tac.constellation.utilities.color.ConstellationColor;
+import org.mockito.MockedStatic;
+import org.mockito.Mockito;
+import static org.mockito.Mockito.doNothing;
+import static org.mockito.Mockito.doReturn;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.mockStatic;
+import static org.mockito.Mockito.spy;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import org.mockito.stubbing.Answer;
+import static org.testng.Assert.assertEquals;
+import static org.testng.Assert.assertFalse;
+import static org.testng.Assert.assertTrue;
+import org.testng.annotations.AfterMethod;
+import org.testng.annotations.BeforeMethod;
+import org.testng.annotations.Test;
+
+/**
+ *
+ * @author Andromeda-224
+ */
+public class AnimationNGTest {
+
+    private String graphId;
+    private ColorWarpAnimation colorWarpAnimation;
+    private DirectionIndicatorAnimation directionIndicatorAnimation;
+    private FlyingAnimation flyingAnimation;
+    private PanAnimation panAnimation;
+    private ThrobbingNodeAnimation throbbingNodeAnimation;
+    private DualGraph graph;
+    private WritableGraph wg;
+    private MockedStatic<ConstellationColor> constyColor;
+    private MockedStatic<AnimationUtilities> animationUtilitiesMocked;
+    private ConstellationColor mockedColor;
+    
+    
+    public AnimationNGTest() {
+    }
+
+
+    @BeforeMethod
+    public void setUpMethod() throws Exception {
+        colorWarpAnimation = spy(ColorWarpAnimation.class);
+        directionIndicatorAnimation = spy(DirectionIndicatorAnimation.class);
+        flyingAnimation = spy(FlyingAnimation.class);
+        throbbingNodeAnimation = spy(ThrobbingNodeAnimation.class);
+        panAnimation = new PanAnimation("Test Name", new Camera(),
+                new Camera(), false);
+
+        graph = new DualGraph(SchemaFactoryUtilities.getSchemaFactory(
+                VisualSchemaFactory.VISUAL_SCHEMA_ID).createSchema());
+        wg =  graph.getWritableGraph("", true);
```

Comment:
should ensure that the you call either `commit()` or `rollback()` at the end so that the write lock is released